custom_components/ramses_cc/climate.py

```diff
@@ -209,11 +214,11 @@ def hvac_action(self) -> str | None:
     def hvac_mode(self) -> str | None:
         """Return the Controller's current operating mode of a Controller."""
 
-        if self._device.system_mode is None:
+        if self.rf_entity.system_mode is None:
             return  # unable to determine
-        if self._device.system_mode[CONF_SYSTEM_MODE] == SystemMode.HEAT_OFF:
+        if self.rf_entity.system_mode[CONF_SYSTEM_MODE] == SystemMode.HEAT_OFF:
             return HVACMode.OFF
-        if self._device.system_mode[CONF_SYSTEM_MODE] == SystemMode.AWAY:
+        if self.rf_entity.system_mode[CONF_SYSTEM_MODE] == SystemMode.AWAY:
```

Comment:
I'll have a look at this file, after `rf_entity` is reverted back to old name - will make thsi PR much smaller.